msu/hooks/items/item_container.nut

```diff
@@ -92,4 +92,36 @@
 
 		return __original(_item);
 	}
+
+	q.getItemsByFunction <- function( _function )
+	{
+		local ret = [];
+
+		foreach (slot, spaces in ::Const.ItemSlotSpaces)
+		{
+			for (local i = 0; i < spaces; i++)
+			{
+				this.__putItemsAtSlotIntoArray(slot, ret);
+			}
```

Comment:
Wait, I don't get it. Why this inner loop? Shouldn't this just run for all the slots, and the private function then iterates over the items in that slot?

Reply:
Fixed